Source/JavaScriptCore/runtime/WriteBarrier.h

```diff
@@ -352,3 +352,24 @@ class WriteBarrierStructureID {
 };
 
 } // namespace JSC
+
+namespace WTF {
+
+template<typename T> struct VectorTraits<JSC::WriteBarrier<T>> : public SimpleClassVectorTraits {
+    static_assert(std::is_trivially_destructible<JSC::WriteBarrier<T>>::value);
+    static const bool canInitializeWithMemset = true;
+    static const bool canCopyWithMemcpy = true;
+};
+
+template<> struct VectorTraits<JSC::WriteBarrier<JSC::Unknown>> : public SimpleClassVectorTraits {
+    static_assert(std::is_trivially_destructible<JSC::WriteBarrier<JSC::Unknown>>::value);
+#if USE(JSVALUE64)
+    // We can memset only in JSVALUE64 since empty value is zero. On the other hand, JSVALUE32_64's empty value is not zero.
+    static const bool canInitializeWithMemset = true;
+#else
+    static const bool canInitializeWithMemset = false;
+#endif
+    static const bool canCopyWithMemcpy = true;
+};
```

Comment:
Can we also set `canCompareWithMemcmp` to `true`? Also seems like `operator==` for two `WriteBarrier<Unknown>` should not decode the two values, just compare them still encoded.

Reply:
Rigt, changed :)

Reply:
Ah, `canCompareWithMemcmp` was true since `SimpleClassVectorTraits` is setting it to true :)